File: Taxonomy/Entity.cs
namespace Taxonomy;

public sealed record Entity(
    Guid Id,
    ISet<EntitySource> Sources,
    ISet<EntityTag> Tags,
    ISet<EntityRelation> Relations
);

public sealed record EntityTag(string Kind, string Value);

public sealed record EntityRelation(string Kind, Guid Other);

public sealed record EntitySource(Source Source, string SourceId);

public sealed record Source(string Name);

File: preonsharp-react-app/src/models/Entity.ts
import {Guid} from './Guid.ts';

export type EntitySource = {
  readonly idNamespace: string;
  readonly sourceId: string;
};

export type EntityTag = {
  readonly kind: string;
  readonly value: string;
}

export type EntityRelation = {
  readonly kind: string;
  readonly other: Guid;
}

export type Entity = {
  readonly id: Guid;
  readonly sources: EntitySource[];
  readonly tags: EntityTag[];
  readonly names: EntityTag[];
  readonly relations: EntityRelation[];
}
